src/client/automation/playback/type/type-text.js

```diff
@@ -84,26 +105,68 @@ function _excludeInvisibleSymbolsFromSelection (selection) {
     return selection;
 }
 
+// NOTE: Typing can be prevented in Chrome/Edge but can not be prevented in IE11 or Firefox
+//       Firefox does not support TextInput event
+//       Safari supports the TextInput event but has a bug: e.data is added to the node value.
+//       So in Safari we need to call preventDefault in the last textInput handler but not prevent the Input event
+
+let forceInputInSafari;
```

Comment:
Usually I don't like nested methods, but in this case I would love to encapsulate logic for Safari in `simulateTextInput` method. We can move `forceInputInSafari`, `onSafariTextInput`, and `onSafariPreventTextInput` in it.
@AndreyBelym